src/scatter_buildblock/ScatterSimulation.cxx

```diff
@@ -843,16 +843,19 @@ ScatterSimulation::downsample_scanner(int new_num_rings, int new_num_dets)
       new_num_dets = this->proj_data_info_sptr->get_scanner_ptr()->get_num_detectors_per_ring();
       approx_num_non_arccorrected_bins = this->proj_data_info_sptr->get_num_tangential_poss();
       // preserve the length of the scanner the following includes gaps
-      float scanner_length_block = new_scanner_sptr->get_num_axial_buckets() * new_scanner_sptr->get_num_axial_blocks_per_bucket()
-                                   * new_scanner_sptr->get_axial_block_spacing();
+      float scanner_length_block
+          = old_scanner_ptr->get_num_axial_buckets() * old_scanner_ptr->get_num_axial_blocks_per_bucket()
+                * old_scanner_ptr->get_axial_block_spacing()
+            - (old_scanner_ptr->get_axial_block_spacing()
+               - old_scanner_ptr->get_num_axial_crystals_per_block() * old_scanner_ptr->get_axial_crystal_spacing());
```

Comment:
I agree with this change. I vote though for adding a member `Scanner::get_axial_length()` with doc like `total axial length covered by the detectors (including any gaps between blocks etc).` (add a `\todo Need to update this function when enabling different spacing between blocks and buckets` or similar). This will make this code much clearer. It will also isolate necessary changes in #1400.

(We could even add a test on `Scanner` that this corresponds to the `(axial_position[list]-axial_position[first) + axial_crystal_spacing)`)

Reply:
change variable name though?

Reply:
that's a good idea - will do!